internal/pkg/cli/project_init.go

```diff
@@ -151,6 +161,42 @@ func (o *initProjectOpts) Execute() error {
 		return err
 	}
 	o.prog.Stop(log.Ssuccessf(fmtDeployProjectComplete, color.HighlightUserInput(o.ProjectName)))
+
+	return o.projectStore.CreateProject(&archer.Project{
+		AccountID: caller.Account,
+		Name:      o.ProjectName,
+		Domain:    o.DomainName,
+	})
+}
+
+func (o *initProjectOpts) validateProject(projectName string) error {
+	if err := validateProjectName(projectName); err != nil {
+		return err
+	}
+	proj, err := o.projectStore.GetProject(projectName)
+	if err != nil {
+		var noSuchProjectErr *store.ErrNoSuchProject
+		if errors.As(err, &noSuchProjectErr) {
+			return nil
+		}
+		return fmt.Errorf("get project %s: %w", projectName, err)
+	}
+	if proj.Domain != o.DomainName {
```

Comment:
I wonder if we can use https://golang.org/pkg/reflect/#DeepEqual to check if the two objects differ

Reply:
I think we should be good. `Project` has projectName (must be the same), accountID (must be the same as well otherwise `GetProject` won't get this project). Not sure if I missed anything?

Reply:
hehe that's because I'm adding `Tags` now as a field to a project :P https://github.com/aws/amazon-ecs-cli-v2/pull/779/files so i wanted that to error out too if it's different. But I can do that change in my PR :)